lib/dao.js

```diff
@@ -3059,7 +3059,10 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
         function replaceCallback(err, data) {
           if (err) return cb(err);
-
+          // if connector is cloudant, add the `_rev` property back before passing it to the user
+          if (connector.name === 'cloudant') {
+            context.data._rev = data._rev;
```

Comment:
IMO database returns `ok` doesn't mean anything, for example, we can modify the data in connector, ideally we know what we are doing, we guarantee we don't pollute user's data, while, but if some code by accident add/delete something, database will still return ok(it doesn't care what your new data is, as long as the replacement operation executes successfully), and juggler still returns user passed in data, finally no place could catch the mistake...

Especially for cloudant(fault-tolerant database), only by checking the returned result we can know what data it actually stores :(